src/frontend/sharedComponents/PermissionAudio.tsx

```diff
@@ -1,72 +1,100 @@
+    id: 'screens.AudioPermission.Button.goToSettings',
+    defaultMessage: 'Go to Settings',
+    description:
+      'Screen button text for navigate user to settings when audio permission was denied',
+  },
+});
+interface PermissionAudioProps {
   closeSheet: () => void;
   isOpen: boolean;
+  sheetRef: React.RefObject<BottomSheetModalMethods>;
 }
 
-export const PermissionAudio: FC<PermissionAudio> = props => {
-  const {sheetRef, closeSheet, isOpen} = props;
+export const PermissionAudio: FC<PermissionAudioProps> = ({
+  closeSheet,
+  isOpen,
+  sheetRef,
+}) => {
   const {formatMessage: t} = useIntl();
-  const [permissionResponse] = Audio.usePermissions({request: false});
+  const navigation = useNavigationFromRoot();
+  const [permissionResponse, setPermissionResponse] =
+    useState<PermissionResponse | null>(null);
 
-  const handlePermissionGranted = useCallback(() => {
+  const handleOpenSettings = () => {
+    Linking.openSettings();
     closeSheet();
-  }, [closeSheet]);
-
-  const isPermissionGranted = Boolean(permissionResponse?.granted);
+  };
 
-  useEffect(() => {
-    if (isPermissionGranted) handlePermissionGranted();
-  }, [isPermissionGranted, handlePermissionGranted]);
+  const handleRequestPermission = async () => {
+    const response = await Audio.requestPermissionsAsync();
+    setPermissionResponse(response);
+    if (response.status === 'granted') {
+      closeSheet();
+      navigation.navigate('Audio');
+    } else if (response.status === 'denied' && response.canAskAgain) {
+      closeSheet();
+    } else if (response.status === 'denied' && !response.canAskAgain) {
+      handleOpenSettings();
+    }
+  };
 
   let onPressActionButton: () => void;
   let actionButtonText: string;
-  switch (permissionResponse?.status) {
-    case undefined:
-    case PermissionStatus.UNDETERMINED:
-      onPressActionButton = handleOpenSettings;
-      actionButtonText = t(m.allowButtonText);
-      break;
-    case PermissionStatus.DENIED:
-      if (permissionResponse.canAskAgain) {
-        onPressActionButton = handleOpenSettings;
-        actionButtonText = t(m.allowButtonText);
-      } else {
-        onPressActionButton = handleRequestPermissions;
-        actionButtonText = t(m.goToSettingsButtonText);
-      }
-      break;
-    case PermissionStatus.GRANTED:
-      onPressActionButton = handlePermissionGranted;
-      actionButtonText = t(m.allowButtonText);
-      break;
-    default:
-      throw new Error('Unexpected permission response');
+
+  if (!permissionResponse) {
+    onPressActionButton = async () => {
+      await handleRequestPermission();
+    };
+    actionButtonText = t(m.allowButtonText);
+  } else if (permissionResponse.status === 'denied') {
+    onPressActionButton = handleOpenSettings;
+    actionButtonText = t(m.goToSettingsButtonText);
+  } else {
+    onPressActionButton = async () => {
+      await handleRequestPermission();
+    };
+    actionButtonText = t(m.allowButtonText);
```

Comment:
this may be a preference thing, so feel free to ignore, but this feels cleaner as 2 seperate ternaries:

```ts
const onPressActionButton = !permissionResponse
  ? async () => {
      await handleRequestPermission();
    }
  : permissionResponse.status === 'denied'
  ? handleOpenSettings
  : async () => {
      await handleRequestPermission();
    };

const actionButtonText= !permissionResponse
  ? t(m.allowButtonText)
  : permissionResponse.status === 'denied'
  ? t(m.goToSettingsButtonText)
  : t(m.allowButtonText);
```